servo-core/src/main/java/com/netflix/servo/monitor/StatsMonitor.java

```diff
@@ -359,6 +415,14 @@ private void updateGauges() {
    */
   @Override
   public List<Monitor<?>> getMonitors() {
```

Comment:
To check my understanding, the expected flow is:

1. User registers StatMonitor, so `getMonitors` will start getting called.
2. At some point they unregister, but the StatMonitor won't get collected because of the reference from the executor.
3. `getMonitors` will no longer be called so now lastUsed will not get updated and it can expire.

The condition is then catching the case where they register a monitor that was previous unregistered?